java/mtsj/core/src/test/java/com/devonfw/application/mtsj/general/common/PermissionCheckTest.java

```diff
@@ -1,27 +1,84 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.domain.Sort.Direction;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 
+import com.devonfw.application.mtsj.SpringBootApp;
+import com.devonfw.application.mtsj.bookingmanagement.common.api.to.BookingSearchCriteriaTo;
+import com.devonfw.application.mtsj.bookingmanagement.common.api.to.InvitedGuestEto;
+import com.devonfw.application.mtsj.bookingmanagement.common.api.to.InvitedGuestSearchCriteriaTo;
+import com.devonfw.application.mtsj.bookingmanagement.common.api.to.TableEto;
+import com.devonfw.application.mtsj.bookingmanagement.common.api.to.TableSearchCriteriaTo;
+import com.devonfw.application.mtsj.bookingmanagement.logic.impl.BookingmanagementImpl;
+import com.devonfw.application.mtsj.dishmanagement.common.api.to.CategoryEto;
+import com.devonfw.application.mtsj.dishmanagement.common.api.to.CategorySearchCriteriaTo;
+import com.devonfw.application.mtsj.dishmanagement.common.api.to.DishCto;
+import com.devonfw.application.mtsj.dishmanagement.common.api.to.DishSearchCriteriaTo;
+import com.devonfw.application.mtsj.dishmanagement.common.api.to.IngredientEto;
+import com.devonfw.application.mtsj.dishmanagement.common.api.to.IngredientSearchCriteriaTo;
+import com.devonfw.application.mtsj.dishmanagement.logic.impl.DishmanagementImpl;
+import com.devonfw.application.mtsj.general.common.impl.security.ApplicationAccessControlConfig;
+import com.devonfw.application.mtsj.imagemanagement.common.api.to.ImageEto;
+import com.devonfw.application.mtsj.imagemanagement.common.api.to.ImageSearchCriteriaTo;
+import com.devonfw.application.mtsj.imagemanagement.logic.impl.ImagemanagementImpl;
+import com.devonfw.application.mtsj.ordermanagement.common.api.to.OrderLineEto;
+import com.devonfw.application.mtsj.ordermanagement.common.api.to.OrderLineSearchCriteriaTo;
+import com.devonfw.application.mtsj.ordermanagement.common.api.to.OrderSearchCriteriaTo;
+import com.devonfw.application.mtsj.ordermanagement.logic.impl.OrdermanagementImpl;
+import com.devonfw.application.mtsj.usermanagement.common.api.to.UserEto;
+import com.devonfw.application.mtsj.usermanagement.common.api.to.UserRoleEto;
+import com.devonfw.application.mtsj.usermanagement.common.api.to.UserRoleSearchCriteriaTo;
+import com.devonfw.application.mtsj.usermanagement.common.api.to.UserSearchCriteriaTo;
+import com.devonfw.application.mtsj.usermanagement.logic.impl.UsermanagementImpl;
 import com.devonfw.module.test.common.base.ModuleTest;
 
 /**
  * Tests the permission check in logic layer.
  *
  */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = SpringBootApp.class)
 public class PermissionCheckTest extends ModuleTest {
```

Comment:
should be `ComponentTest` instead of `ModuleTest`. Then we can also remove `@RunWith` annotation.